tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html

```diff
@@ -1,252 +0,0 @@
-              >.
-            </p>
-
-            <p>
-              If you think TensorBoard is configured properly, please see
-              <a
-                href="https://github.com/tensorflow/tensorboard/blob/master/README.md#my-tensorboard-isnt-showing-any-data-whats-wrong"
-                >the section of the README devoted to missing data problems</a
-              >
-              and consider filing an issue on GitHub.
-            </p>
-          </div>
-        </template>
-        <template is="dom-if" if="[[!_dataNotFound]]">
-          <tf-tag-filterer tag-filter="{{_tagFilter}}"></tf-tag-filterer>
-          <template is="dom-repeat" items="[[_categories]]" as="category">
-            <tf-category-paginated-view
-              category="[[category]]"
-              initial-opened="[[_shouldOpen(index)]]"
-            >
-              <template>
-                <tf-histogram-loader
-                  run="[[item.run]]"
-                  tag="[[item.tag]]"
-                  active="[[active]]"
-                  tag-metadata="[[_tagMetadata(_runToTagInfo, item.run, item.tag)]]"
-                  time-property="[[_timeProperty]]"
-                  histogram-mode="[[_histogramMode]]"
-                  request-manager="[[_requestManager]]"
-                ></tf-histogram-loader>
-              </template>
-            </tf-category-paginated-view>
-          </template>
-        </template>
-      </div>
-    </tf-dashboard-layout>
-
-    <style include="dashboard-style"></style>
-    <style>
-      .no-data-warning {
-        max-width: 540px;
-        margin: 80px auto 0 auto;
-      }
-    </style>
-  </template>
-
-  <script>
-    'use strict';
-
-    Polymer({
-      is: 'tf-histogram-dashboard',
-      properties: {
-        reloadOnReady: {
-          type: Boolean,
-          value: true,
-        },
-        _histogramMode: {
-          type: String,
-          value: 'offset',
-        },
-        _timeProperty: {
-          type: String,
-          value: 'step',
-        },
-
-        _selectedRuns: Array,
-        _runToTag: Object, // map<run: string, tags: string[]>
-        _runToTagInfo: Object,
-        _dataNotFound: Boolean,
-        _tagFilter: String,
-        _restamp: {
-          type: Boolean,
-          value: false,
-        },
-
-        // Categories must only be computed after _dataNotFound is found to be
-        // true and then polymer DOM templating responds to that finding. We
-        // thus use this property to guard when categories are computed.
-        _categoriesDomReady: Boolean,
-        _categories: {
-          type: Array,
-          computed:
-            '_makeCategories(_runToTag, _selectedRuns, _tagFilter, _categoriesDomReady)',
-        },
-
-        _requestManager: {
-          type: Object,
-          value: () => new tf_backend.RequestManager(),
-        },
-      },
-
-      listeners: {
-        'content-visibility-changed': '_redrawCategoryPane',
```

Comment:
The auto-migration script doesn't cover `listeners`, so we should add something like

`this.addEventListener('content-visibility-changed', this._redrawCategoryPane.bind(this))`
in either the `ready()` or `connectedCallback()`

Reply:
We looked into this a bit. This isn’t a standard DOM event; rather, it’s
a custom Polymer event that was originally added in #1465. But there
isn’t any code to actually fire the event; `git log -S` shows that it
was removed in #2273 via #2392. Thus, while the absence of this listener
may be a bug, it’s not new, so I’ll keep this migration patch as is to
make things simple. Thanks for the keen eye, though!